core/src/main/scala/dagr/core/tasksystem/Pipe.scala

```diff
@@ -139,8 +137,11 @@ object Pipes {
       */
     override def applyResources(resources: ResourceSet): Unit = {
       val (fixed, variable) = partition
-      var remainingResource = resources
-      fixed.foreach(task => { task.applyResources(task.resources); remainingResource -= task.resources })
+      val remainingResource = resources - fixed.map( task => {
```

Comment:
I get that this eliminates a var, but it also goes from 2 to 5 lines of code, and I find it quite a bit less readable.